### PR TITLE
Hash introduce to compare docs and early exit in case already exists

### PR DESCRIPTION
**Pull request by rednitish**

### CODE REVIEW

**Comment by xdavidson** [comment]
Nitesh,

Please add some unit tests for the new servlets.

Lisa

**Comment by rednitish** [comment]
Lisa, I have added test cases on the enhancements done, can you please refview ?

**Comment by rednitish** [comment]
@xdavidson  I have improved test cases now. Can you please review ?

File: pantheon-bundle/src/test/java/com/redhat/pantheon/servlet/module/ModuleVersionUploadTest.java [hunk 1]
```diff
@@ -280,6 +280,139 @@ void uploadIdenticalDraftVersion() throws Exception {
...
+        );
+        assertEquals("This is the draft html content",
+                module.getDraftContent(LocaleUtils.toLocale("es_ES")).get().cachedHtml().get().data().get()
+        );
+        assertEquals("This is the released adoc content",
+                module.getReleasedContent(LocaleUtils.toLocale("es_ES")).get().asciidocContent().get()
+        );
+        verify(asciidoctorService, never()).getModuleHtml(
+                any(ModuleVersion.class), any(Resource.class), anyMap(), anyBoolean());
+    }        @Test
```
Comment (carlosmunoz) [inline pantheon-bundle/src/test/java/com/redhat/pantheon/servlet/module/ModuleVersionUploadTest.java:391]: ooff... this formating :nauseated_face:

Comment (rednitish) [reply pantheon-bundle/src/test/java/com/redhat/pantheon/servlet/module/ModuleVersionUploadTest.java]: Auto formatting sometimes does  strange things :)

**Comment by rednitish** [comment]
@xdavidson  Fixed formatting now, Can you please reveiew ?

**Comment by xdavidson** [comment]
> @xdavidson Fixed formatting now, Can you please reveiew ?

@rednitish, can you bring the test coverage to green? According to the report, we are still down by -0.04%

**Comment by rednitish** [comment]
Coverage

@xdavidson  I have already improved the coverage as you can check in attached image,

**Comment by xdavidson** [comment]
> Coverage
> 
> @xdavidson I have already improved the coverage as you can check in attached image,
@rednitish, you did improve the coverage!
according to the report you provided, there is no negative impact on the coverage

Thank you for sharing the report. 

Lisa

**Pull request merged**